pkg/validation/user.go

```diff
@@ -85,3 +87,32 @@ func ValidateUserUpdate(oldUser, newUser *kubermaticv1.User) field.ErrorList {
 
 	return allErrs
 }
+
+func ValidateUserDelete(ctx context.Context, user *kubermaticv1.User, client ctrlruntimeclient.Client) error {
+	projects, err := kubermaticv1helper.GetUserOwnedProjects(ctx, client, user.Name)
+	if err != nil {
+		return err
+	}
+
+	for _, project := range projects {
+		// if project has multiple owner users
+		if len(project.OwnerReferences) > 1 {
+			continue
+		}
+
+		// project has single owner user then check if project has resources
+		hasClusters, err := kubermaticv1helper.HasClusters(ctx, client, project.Name)
+		if err != nil {
+			return err
+		}
```

Comment:
This is not possible, Cluster objects live on seed clusters, users live on the master.

Reply:
oh right!
I can try to get seed client then to fetch the clusters, right?